RevenueCatUI/Templates/Components/TemplateComponentsView.swift

```diff
@@ -12,20 +12,36 @@ import SwiftUI
 @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
 struct TemplateComponentsView: View {
 
-    @Environment(\.locale)
-    var locale
-
     let paywallComponentsData: PaywallComponentsData
+    let componentViewModels: [PaywallComponentViewModel]
 
-    init(paywallComponentsData: PaywallComponentsData) {
+    init(paywallComponentsData: PaywallComponentsData, locale: Locale, offering: Offering) {
         self.paywallComponentsData = paywallComponentsData
+
+        let components = paywallComponentsData.componentsConfig.components
+        self.componentViewModels = components.map { component in
+
+            // STEP 1 - Get list of preferred locales (and default)
+
+            // STEP 2 - Get localization for one of preferred locales in order
+            // TOOD: This logic is so wrong
+            let localizations = paywallComponentsData.componentsLocalizations
+            let localization = localizations[locale.identifier] ?? localizations.values.first ?? [String:String]()
+
+            // Step 3 - Validate all variables are supported in localization
+
+            // Step 3.5 - Validate all packages needed exist (????)
+
+            // Step 4 - Make the view models
+            return component.toViewModel(offering: offering, locale: locale, localization: localization)
```

Comment:
Note to any other reviewers: This is where we will do a validation and these comments are essentially TODO so we can keep that in mind when coming back to this area in near future 😊